src/main/java/com/j256/ormlite/field/types/LocalDateTimeSqlType.java

```diff
@@ -0,0 +1,53 @@
+package com.j256.ormlite.field.types;
+
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import com.j256.ormlite.field.FieldType;
+import com.j256.ormlite.field.SqlType;
+import com.j256.ormlite.misc.SqlExceptionUtil;
+import com.j256.ormlite.support.DatabaseResults;
+
+/**
+ * A custom persister that is able to store the java.time.LocalDateTime class in the database as Timestamp object.
+ * This class should be used only when database used does not support JDBC 4.2, since it converts java.time.LocalDateTime
+ * to java.sql.Timestamp.
+ *
+ * @author graynk
+ */
+public class LocalDateTimeSqlType extends LocalDateTimeType {
+
+    private static final LocalDateTimeSqlType singleton = isJavaTimeSupported() ? new LocalDateTimeSqlType() : null;
+    public static LocalDateTimeSqlType getSingleton() { return singleton; }
+    private LocalDateTimeSqlType() { super(SqlType.LOCAL_DATE_TIME); }
+    protected LocalDateTimeSqlType(SqlType sqlType, Class<?>[] classes) { super(sqlType, classes); }
+
+    @Override
+    public Object parseDefaultString(FieldType fieldType, String defaultStr) throws SQLException {
+        try {
+            return Timestamp.valueOf(LocalDateTime.parse(defaultStr, DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss[.SSS]")));
```

Comment:
This caused problems in the tests for me. A better approach here (and ultimately more user friendly) would be variable length second fractions.

An example of this is here: https://github.com/Bo98/ormlite-core/commit/00fed82b928ef4e2e5b6aad1d8ce54ddeccffc3b, along with adjustments to the tests to make sure they are consistently truncated.